crates/geoip/src/resolver.rs

```diff
@@ -0,0 +1,29 @@
+use std::{net::IpAddr, sync::Arc};
+
+pub use {
+    crate::maxminddb::geoip2::City,
+    local::LocalResolver,
+    maxmind::{MaxMindResolver, MaxMindResolverError},
+};
+
+mod local;
+mod maxmind;
+
+#[derive(Debug, Clone)]
+pub struct GeoData {
+    pub continent: Option<Arc<str>>,
+    pub country: Option<Arc<str>>,
+    pub region: Option<Vec<String>>,
+    pub city: Option<Arc<str>>,
+}
+
+pub trait GeoIpResolver: Clone {
```

Comment:
We may just call it `Resolver` and reexport it in `lib.rs`, so one can easily import it as `geoip::Resolver`